Automattic/jetpack
# Classic Theme Helper: Copy Social Links code to Classic Theme Helper package

Fixes https://github.com/Automattic/vulcan/issues/431

## Proposed changes:

* Moved the main file `social-links.php`.
* In order to avoid any issues if the package file happens to load just before the module file (as the package file is being read in my local tests), I've commented out the class initialization from the package file for now, and will modify this when requiring the package in the next PR.
* Added some Phan suppressions and attempted some Phan suggested fixes related to adding class / function checks, but these won't be tested until requiring this feature via the package (in the next PR).
* There is no WebPack build added, as there is none in the Jetpack plugin either for this feature.

### Other information:


## Jetpack product discussion

https://github.com/Automattic/vulcan/issues/431

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* Technically the package isn't being used yet, and no build is happening, so a code check is the main thing.
* However in testing I did come across a fatal and had to make a few changes to make sure the package version didn't try to initialize the class as the file was being read, so we should make sure functionality is not affected. To do so:
  
Self-hosted and Atomic:
* On a test site, add a theme that includes support for Jetpack's social-links. The only theme I could find is 'Syntax'. 
* Otherwise, make sure you have a classic (not block) theme, such as Twenty Nineteen, and add the following snippet with a functionality plugin (like 'Code Snippets'):
```
add_action( 'after_setup_theme', function () {
	add_theme_support( 'social-links', array(
		'facebook', 'twitter', 'linkedin', 'tumblr',
	) );
} );
```
* Enable Social sharing - `/wp-admin/admin.php?page=jetpack#/sharing` and add a connection. 
* You can add a Facebook page by using the Gmail username and pw for `credsflow....` in the SS (internal only).
* Visit the Customizer, and you should see a 'Connect' menu item, that allows you to select your newly added connection.
* This should work both on trunk and with this PR applied.
* On Atomic make sure to test with the PR checked out for both Jetpack and WPcom Features, and each individually.

Simple:
* Add a social connection at `https://wordpress.com/marketing/connections/yoursite.com`
* Add a theme which includes support for Jetpack's social-links - Syntax for example.
* Visit the Customizer and you should see the 'Connect' menu item allowing you to connect your new social connection.
* This should work both with and without the patches applied - see testing comment below (and also one at a time to replicate Jetpack only and Mu Wpcom only).
